src/util/requestWrapper.ts

```diff
@@ -77,6 +80,16 @@ export default class RequestUtil {
         await this.syncTime();
       }
 
+      // Optional, set to 5000 by default. Increase if timestamp/recv_window errors are seen.
+      if (this.options.recv_window) {
+
+        if (endpoint.includes("spot/")) { // lucky us. only the spot API uses a different param for this
+          params = { ...params, recvWindow: this.options.recv_window };
+        } else {
+          params = { ...params, recv_window: this.options.recv_window }
+        }
+      }
```

Comment:
Yikes, if only they were consistent... good catch

Reply:
I just realised that the my bybit client classes aren't using the BaseRestClient yet, which explains why you've been using the requestWrapper too.

All my connectors (bybit, ftx, binance) follow similar design patterns. Binance is my most recent and probably more improved iteration of these patterns, where I decided to move all request logic to BaseRestClient, which is extended by the REST client. This makes the dependency on the getTime endpoint less circular and solved a bunch of other problems. Eventually I wanted to migrate to that design with this bybit and also the ftx connector.

You can see the implementation here:
https://github.com/tiagosiebler/binance/blob/master/src/main-client.ts#L90-L107

Could you migrate the request logic into the BaseRestClient in this PR? It doesn't need to be a full migration on the inverse/linear/futures clients yet, that can come separately since that's probably no small feat.

I would start with migrating what's currently in the bybit-api requestWrapper into the BaseRequestClient, then apply your improvements to it to also support signed requests to the spot apis, then update the spot client to use the BaseRequestClient similarly to how the REST clients in my binance connector do it.

Then when it comes to later migrating the other bybit-api REST clients to extend the BaseRequestClient it'll be less work, since it has an extension of the logic already in the requestWrapper. Does this make sense? Feel free to ping me on telegram if you have questions/thoughts/concerns.